client/pages/offer-thread.jsx

```diff
@@ -99,6 +126,10 @@ class OfferThread extends React.Component {
                     }
                   )
                       }
+                  <div
+                  style={{ float: 'left', clear: 'both' }}
+                  ref={el => { this.messagesEnd = el; }}>
```

Comment:
This is the old way of using refs. Now we do this with `React.createRef()` (like in our `react-file-uploads` exercise).

 


```jsx
constructor(props) {
   //... blah blah
  this.chatDivRef = React.createRef();
}
// ...
componentDidUpdate(prevProps, prevState) {
  if (this.state.messageIn !== prevState.messageIn) {
    this.chatDivRef.current.scrollTop = this.chatDivRef.scrollHeight;
  }
}
// ...
render() {
  //...
  <div className="chat" ref={this.chatDivRef}>
  // ...
}
```